app/integration/runtime/src/main/java/io/syndesis/integration/runtime/logging/IntegrationLoggingListener.java

```diff
@@ -31,27 +31,25 @@ public IntegrationLoggingListener(ActivityTracker tracker) {
 
     @Override
     public String onLog(Exchange exchange, CamelLogger camelLogger, String message) {
-        if (tracker == null) {
-            return message;
-        }
-
-        final String activityId = ActivityTracker.getActivityId(exchange);
-        if (activityId != null) {
-            final Marker marker = camelLogger.getMarker();
-            final String step = marker != null ? marker.getName() : "null";
-
-            final Message in = exchange.getIn();
-            String stepTrackerId = in.getHeader(IntegrationLoggingConstants.STEP_TRACKER_ID, String.class);
-            if( stepTrackerId == null ) {
-                stepTrackerId = KeyGenerator.createKey();
+        if (tracker != null) {
```

Comment:
Prefer early exit
```suggestion
        if (tracker == null) {
        	return message;
        }
```

Reply:
That was one of the things detected as code smell because you had two different returns for the same value.

Reply:
Well, that smell was wrong. You could do:
```
        if (tracker == null || ActivityTracker.getActivityId(exchange) == null) {
            return message;
        }
```

to make it happy, but that leads to calling `ActivityTracker.getActivityId(exchange)` twice in happy day scenario.

Reply:
I will just remove the change then :)